src/main/scala/com/rbmhtechnology/calliope/StoredEvent.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2015 - 2017 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.calliope
+
+import java.time.Instant
+
+case class StoredEvent(sequenceNr: Long, creationTimestamp: Instant, topic: String, aggregateId: String, event: Array[Byte])
```

Comment:
+1 to parameter name alignment

Regarding the specialization - `SequencedMessage` is a broader concept to publish arbitrary payloads that have a notion of sequencing which can be used by many producers (transactional or not) whereas the `StoredEvent` is a more specific type only used for transactional message processing.
The properties of both classes do not align as the `SequencedMessage` does contain the source-id (which is not needed in the StoredEvent as it is set by the TransactionalEventProducer. On the other hand the `StoredEvent` contains the target-topic which is not needed by the `SequencedMessage`.